src/Autonomous.cpp

```diff
@@ -95,6 +95,34 @@ double dist(const pose_t &p1, const pose_t &p2, double theta_weight)
 	return diff.norm();
 }
 
+bool Autonomous::currentLegDone()
+{
+	return nav_state == NavState::DONE;
+}
+
+bool Autonomous::hasNextLeg()
+{
+	return urc_targets.size() > 0;
+}
+
+void Autonomous::startNextLeg()
+{
+	if (!currentLegDone())
+	{
+		log(LOG_INFO, "Cannot start next leg: current leg not done\n");
```

Comment:
Since this is an unexpected case, this should probably be `LOG_WARN` instead of just `LOG_INFO`.